ipa-core/src/protocol/hybrid/agg.rs

```diff
@@ -0,0 +1,255 @@
+use std::collections::BTreeMap;
+
+use futures::{stream, StreamExt, TryStreamExt};
+
+use crate::{
+    error::Error,
+    ff::{boolean::Boolean, boolean_array::BooleanArray, ArrayAccess},
+    helpers::TotalRecords,
+    protocol::{
+        boolean::step::EightBitStep,
+        context::{
+            dzkp_validator::{validated_seq_join, DZKPValidator, TARGET_PROOF_SIZE},
+            Context, DZKPUpgraded, MaliciousProtocolSteps, ShardedContext, UpgradableContext,
+        },
+        hybrid::step::{AggregateReportsStep, HybridStep},
+        ipa_prf::boolean_ops::addition_sequential::integer_add,
+        BooleanProtocols,
+    },
+    report::hybrid::{AggregateableHybridReport, PrfHybridReport},
+    secret_sharing::replicated::semi_honest::AdditiveShare as Replicated,
+};
+
+enum MatchEntry<BK, V>
```

Comment:
we shouldn't require `Empty` state, it can be modelled through the btreemap itself